### PR TITLE
Add static analysis CI job with format check

### PR DESCRIPTION
**Pull request by amdkila**

### CODE REVIEW

**Comment by saadrahim** [comment]
Planning to add clang-format to checks to ensure consistent code formatting. Please take a look and see if there are any objections.

**Comment by saadrahim** [comment]
@amdkila Do we need to add the .clang-format file for customization or are we ok with defaults?

**Comment by amdkila** [comment]
The defaults should be fine I think- the most cotntroversial is the SortIncludes, but that doesn't break functionality for rocRAND (I tested building/packaging/running tests locally). And sure, I'll review the updated githooks when your PR is up

File: .jenkins/staticanalysis.groovy [hunk 1]
```diff
@@ -11,7 +11,7 @@ def runCompileCommand(platform, project, jobName, boolean debug=false)
     def command = """#!/usr/bin/env bash
             set -ex
             cd ${project.paths.project_build_prefix}
-            doxygen docs/Doxyfile
+            doxygen doc/Doxyfile
```
Comment (saadrahim) [inline .jenkins/staticanalysis.groovy:14]: I would rename the folder to docs. All other libraries it is called docs. Just some standardization to make life easier.

**Comment by saadrahim** [comment]
Solves #140

**Comment by saadrahim** [comment]
inactivity.

**Pull request closed without merging**